test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/NoDepsAppTests.cs

```diff
@@ -2,132 +2,63 @@
-                }
-
-                // Check that the first child process is no longer running
-                Waiters.WaitForFileToBeReadable(scenario.StatusFile, _defaultTimeout);
-                var ids = File.ReadAllLines(scenario.StatusFile);
-                var firstProcessId = int.Parse(ids[0]);
-                Waiters.WaitForProcessToStop(
-                    firstProcessId,
-                    TimeSpan.FromSeconds(1),
-                    expectedToStop: true,
-                    errorMessage: $"PID: {firstProcessId} is still alive");
-            }
+            // first app should have shut down
+            Assert.Throws<ArgumentException>(() => Process.GetProcessById(pid));
         }
 
         [Fact]
-        public void RestartProcessThatTerminatesAfterFileChange()
+        public async Task RestartProcessThatTerminatesAfterFileChange()
         {
-            using (var scenario = new NoDepsAppScenario(_logger))
-            {
-                // Wait for the process to start
-                using (var wait = new WaitForFileToChange(scenario.StartedFile))
-                {
-                    scenario.RunDotNetWatch(new[] { "run", scenario.StatusFile });
-
-                    wait.Wait(_defaultTimeout,
-                        expectedToChange: true,
-                        errorMessage: $"File not created: {scenario.StartedFile}");
-                }
-
-                // Then wait for the app to exit
-                Waiters.WaitForFileToBeReadable(scenario.StartedFile, _defaultTimeout);
-                var ids = File.ReadAllLines(scenario.StatusFile);
-                var procId = int.Parse(ids[0]);
-                Waiters.WaitForProcessToStop(
-                    procId,
-                    _defaultTimeout,
-                    expectedToStop: true,
-                    errorMessage: "Test app did not exit");
-
-                // Then wait for it to restart when we change a file
-                using (var wait = new WaitForFileToChange(scenario.StartedFile))
-                {
-                    // On Unix the file write time is in 1s increments;
-                    // if we don't wait, there's a chance that the polling
-                    // watcher will not detect the change
-                    Thread.Sleep(1000);
-
-                    var fileToChange = Path.Combine(scenario.TestAppFolder, "Program.cs");
-                    var programCs = File.ReadAllText(fileToChange);
-                    File.WriteAllText(fileToChange, programCs);
-
-                    wait.Wait(_defaultTimeout,
-                        expectedToChange: true,
-                        errorMessage: $"Process did not restart because {scenario.StartedFile} was not changed");
-                }
-            }
+            await _app.StartWatcher().OrTimeout();
+            var pid = await _app.GetProcessId().OrTimeout();
+            await _app.HasExited().OrTimeout(); // process should exit after run
```

Comment:
Remind me - the process is not running after this line until the file is changed when watcher will restart the process and let it exit - is this correct?